explorer/Explorer.tsx

```diff
@@ -341,130 +343,270 @@ export class Explorer
+        const { decisionMatrix, grapherConfig } = this.explorerProgram
+        const { grapherId } = grapherConfig
+        const yIndicatorIdsColumn = decisionMatrix.table.get(
+            GrapherGrammar.yIndicatorIds.keyword
+        )
+        if (grapherId && isNotErrorValue(grapherId))
+            return ExplorerChartCreationMode.FromGrapherId
+        if (yIndicatorIdsColumn.numValues)
+            return ExplorerChartCreationMode.FromIndicatorIds
+        return ExplorerChartCreationMode.FromExplorerTableColumnSlugs
+    }
+
+    @action.bound private updateGrapherFromExplorer() {
+        switch (this.chartCreationMode) {
+            case ExplorerChartCreationMode.FromGrapherId:
+                this.updateGrapherFromExplorerUsingGrapherId()
+                break
+            case ExplorerChartCreationMode.FromIndicatorIds:
+                this.updateGrapherFromExplorerUsingIndicatorIds()
+                break
+            case ExplorerChartCreationMode.FromExplorerTableColumnSlugs:
+                this.updateGrapherFromExplorerUsingColumnSlugs()
+                break
+        }
+    }
+
+    @action.bound private updateGrapherFromExplorerCommon() {
         const grapher = this.grapher
         if (!grapher) return
-        const grapherConfigFromExplorer = this.explorerProgram.grapherConfig
         const {
-            grapherId,
-            tableSlug,
-            yIndicatorIds = "",
-            xIndicatorId,
-            colorIndicatorId,
-            sizeIndicatorId,
             yScaleToggle,
             yAxisMin,
             facetYDomain,
             relatedQuestionText,
             relatedQuestionUrl,
             mapTargetTime,
-        } = grapherConfigFromExplorer
-        const grapherConfigFromExplorerOnlyGrapherProps = omit(
-            grapherConfigFromExplorer,
-            [
-                "yIndicatorIds",
-                "xIndicatorId",
-                "colorIndicatorId",
-                "sizeIndicatorId",
-                "mapTargetTime",
+        } = this.explorerProgram.grapherConfig
+
+        grapher.yAxis.canChangeScaleType = yScaleToggle
+        grapher.yAxis.min = yAxisMin
+        if (facetYDomain) {
+            grapher.yAxis.facetDomain = facetYDomain
+        }
+        if (relatedQuestionText && relatedQuestionUrl) {
+            grapher.relatedQuestions = [
+                { text: relatedQuestionText, url: relatedQuestionUrl },
             ]
+        }
+        if (mapTargetTime) {
+            grapher.map.time = mapTargetTime
+        }
+        grapher.slug = this.explorerProgram.slug
+        if (!grapher.id) grapher.id = 0
+    }
+
+    @computed private get columnDefsNotLinkedToTableByIdOrSlug(): Record<
+        number | string,
+        OwidColumnDef
+    > {
+        const { columnDefsNotLinkedToTable } = this.explorerProgram
+        return keyBy(
+            columnDefsNotLinkedToTable as OwidColumnDef[],
```

Comment:
The typecast shouldn't be necessary?
```suggestion
            columnDefsNotLinkedToTable,
```